api/cloudspecificextension/templates_api.go

```diff
@@ -0,0 +1,141 @@
+package cloudspecificextension
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/ingrammicro/cio/api/types"
+	"github.com/ingrammicro/cio/utils"
+	log "github.com/sirupsen/logrus"
+)
+
+// CloudSpecificExtensionTemplateService manages cloud specific extension template operations
+type CloudSpecificExtensionTemplateService struct {
+	concertoService utils.ConcertoService
+}
+
+// NewCloudSpecificExtensionTemplateService returns a Concerto cloud specific extension template service
+func NewCloudSpecificExtensionTemplateService(concertoService utils.ConcertoService) (*CloudSpecificExtensionTemplateService, error) {
+	if concertoService == nil {
+		return nil, fmt.Errorf("must initialize ConcertoService before using it")
+	}
+
+	return &CloudSpecificExtensionTemplateService{
+		concertoService: concertoService,
+	}, nil
+}
+
+// ListTemplates returns the list of cloud specific extension templates as an array of CloudSpecificExtensionTemplate
+func (csets *CloudSpecificExtensionTemplateService) ListTemplates() (templates []*types.CloudSpecificExtensionTemplate, err error) {
+	log.Debug("ListTemplates")
+
+	data, status, err := csets.concertoService.Get("/cse/templates")
+	if err != nil {
+		return nil, err
+	}
+
+	if err = utils.CheckStandardStatus(status, data); err != nil {
+		return nil, err
+	}
+
+	if err = json.Unmarshal(data, &templates); err != nil {
+		return nil, err
+	}
+
+	return templates, nil
+}
+
+// GetTemplate returns a cloud specific extension template by its ID
+func (csets *CloudSpecificExtensionTemplateService) GetTemplate(templateID string) (template *types.CloudSpecificExtensionTemplate, err error) {
+	log.Debug("GetTemplate")
+
+	data, status, err := csets.concertoService.Get(fmt.Sprintf("/cse/templates/%s", templateID))
+	if err != nil {
+		return nil, err
+	}
+
+	if err = utils.CheckStandardStatus(status, data); err != nil {
+		return nil, err
+	}
+
+	if err = json.Unmarshal(data, &template); err != nil {
+		return nil, err
+	}
+
+	return template, nil
+}
+
+// CreateTemplate creates a cloud specific extension template
+func (csets *CloudSpecificExtensionTemplateService) CreateTemplate(templateVector *map[string]interface{}) (template *types.CloudSpecificExtensionTemplate, err error) {
```

Comment:
templateVector is a bad name, what about templateParams?

Reply:
Done, additionally this should be done in more cases (added a note at issue #31 ).